packages/fuels-accounts/src/provider.rs

```diff
@@ -197,24 +197,29 @@ impl Provider {
     }
 
     /// Sends a transaction to the underlying Provider's client.
-    pub async fn send_transaction_and_await_commit<T: Transaction>(&self, tx: T) -> Result<TxId> {
-        let tx_id = self.send_transaction(tx.clone()).await?;
-        let _status = self.client.await_transaction_commit(&tx_id).await?;
-
+    pub async fn send_transaction_and_await_commit<T: Transaction>(
```

Comment:
IMO this function should return a `TxStatus`. We have the status here and the user can calculate the `TxId` if he needs it. Also if we make this change in our `call_or_simulate` we could use only one call instead of using two:
```rust
let tx_id = self.provider.send_transaction_and_await_commit(tx).await?;
self.provider.tx_status(&tx_id).await?
```
This I how I recommend we write the fn:
```rust
pub async fn send_transaction_and_await_commit<T: Transaction>(
    &self,
    mut tx: T,
) -> Result<TxStatus> {
    self.prepare_transaction_for_sending(&mut tx).await?;
    let tx_status = self
        .client
        .submit_and_await_commit(&tx.clone().into())
        .await?
        .into();

    #[cfg(feature = "coin-cache")]
    if matches!(
        tx_status,
        TxStatus::SqueezedOut { .. } | TxStatus::Revert { .. }
    ) {
        self.cache.lock().await.remove_items(tx.used_coins())
    }

    Ok(tx_status)
}
```
There were also some small changes in other parts of the codebase and I have fixed them on this branch: `hal3e/refactor-await-commit-api`

Reply:
Thanks for changing it. This is now a breaking change so please do not forget to add this to the PR description.